build/azure-pipelines.yml

```diff
@@ -1,277 +1,316 @@
-              inlineScript: |
-                  Write-Host "Working folder: $pwd"
-                  $ubuild = build\build.ps1 -get -continue
-
-                  $ubuild.VerifyNuGet()
-                  $ubuild.PackageNuGet()
-
-        - task: CopyFiles@2
-          displayName: 'Copy NuPkg Files to Staging'
-          inputs:
-              SourceFolder: build.out
-              Contents: '*.nupkg'
-              TargetFolder: '$(build.artifactstagingdirectory)'
-              CleanTargetFolder: true
-
-        - task: PublishBuildArtifacts@1
-          displayName: 'Publish NuPkg Files'
-          inputs:
-              PathtoPublish: '$(build.artifactstagingdirectory)'
-              ArtifactName: nupkg
-
-        - task: CopyFiles@2
-          displayName: 'Copy Log Files to Staging'
-          inputs:
-              SourceFolder: build.tmp
-              Contents: '*.log'
-              TargetFolder: '$(build.artifactstagingdirectory)'
-              CleanTargetFolder: true
-          condition: succeededOrFailed()
-
-        - task: PublishBuildArtifacts@1
-          displayName: 'Publish Log Files'
-          inputs:
-              PathtoPublish: '$(build.artifactstagingdirectory)'
-              ArtifactName: logs
-          condition: succeededOrFailed()
+    - stage: Unit_Tests
+      displayName: Unit Tests
+      dependsOn: []
+      jobs:
+          - job: Linux_Unit_Tests
+            displayName: Linux
+            pool:
+                vmImage: ubuntu-latest
+            steps:
+                - task: UseDotNet@2
+                  displayName: Use .Net Core sdk 5.x
+                  inputs:
+                      version: 5.x
+                - task: DotNetCoreCLI@2
+                  displayName: dotnet test
+                  inputs:
+                      command: test
+                      projects: '**/*.Tests.UnitTests.csproj'
+          - job: MacOS_Unit_Tests
+            displayName: Mac OS
+            pool:
+                vmImage: macOS-latest
+            steps:
+                - task: UseDotNet@2
+                  displayName: Use .Net Core sdk 5.x
+                  inputs:
+                      version: 5.x
+                - task: DotNetCoreCLI@2
+                  displayName: dotnet test
+                  inputs:
+                      command: test
+                      projects: '**/*.Tests.UnitTests.csproj'
+          - job: Windows_Unit_Tests
+            displayName: Windows
+            pool:
+                vmImage: windows-latest
+            steps:
+                - task: UseDotNet@2
+                  displayName: Use .Net Core sdk 5.x
+                  inputs:
+                      version: 5.x
+                - task: DotNetCoreCLI@2
+                  displayName: dotnet test
+                  inputs:
+                      command: test
+                      projects: '**/*.Tests.UnitTests.csproj'
+    - stage: Integration_Tests
+      dependsOn: []
```

Comment:
Oh very minor thing I noticed, but Integration_Tests has no display name, which I think is causing it to show up as Integration_Tests Windows/Linux